modules/javafx.web/src/test/java/test/javafx/scene/web/FileTest.java

```diff
@@ -0,0 +1,94 @@
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package test.javafx.scene.web;
+
+import com.sun.javafx.webkit.UIClientImplShim;
+import com.sun.webkit.WebPage;
+import com.sun.webkit.WebPageShim;
+import javafx.concurrent.Worker.State;
+import javafx.scene.web.WebEngineShim;
+import netscape.javascript.JSObject;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.util.Base64;
+import java.util.concurrent.CountDownLatch;
+
+import static javafx.concurrent.Worker.State.SUCCEEDED;
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+public class FileTest extends TestBase {
+    private final WebPage page = WebEngineShim.getPage(getEngine());
+    private String[] fileList = { new File("src/test/resources/test/html/HelloWorld.txt").getAbsolutePath() };
+
+    private State getLoadState() {
+        return submit(() -> getEngine().getLoadWorker().getState());
+    }
+
+    private String getScriptString(String api) {
+        String scriptContent = String.format("<script type='text/javascript'>" +
+                                    "var result;" +
+                                    "window.addEventListener('click', (e) => {" +
+                                        "document.getElementById('file').click();" +
+                                    "});" +
+                                    "function readFile()" +
+                                    "{" +
+                                        "file = event.target.files[0];" +
+                                        "result = file." + api + ";" +
+                                    "}" +
+                               "</script>" +
+                               "<body> <input type='file' id='file' onchange='readFile()'/></body>");
+        return scriptContent;
+    }
+
+    @Before
+    public void before() {
+        UIClientImplShim.test_setChooseFiles(fileList);
+    }
+
+    private void loadFileReaderTestScript(String testScript) {
+        loadContent(testScript);
+        submit(() -> {
+            // we send a dummy mouse click event at (0,0) to simulate click on file chooser button.
+            WebPageShim.click(page, 0, 0);
+        });
+    }
+
+    @Test public void testFileName() {
+        loadFileReaderTestScript(getScriptString("name"));
+        submit(() -> {
+            assertEquals("Unexpected file name received", "HelloWorld.txt", getEngine().executeScript("window.result"));
+        });
+    }
+}
```

Comment:
The test requires a few changes:
1. Correct copyright year
2. Several imports can be replaced.
3. method private State getLoadState() is unused
4. method getScriptString() can be replaced with a string member variable
5. Line 70,71 needs format correction

I tried these changes locally, attaching the file with required changes. Please check
[FileTest.java.zip](https://github.com/user-attachments/files/16813919/FileTest.java.zip)

Reply:
I have taken the requested changes from your file.
This was simply taken from test.javafx.scene.web.FileReaderTest with future tests for other file properties in mind.